### PR TITLE
feat: Read motion sensor

### LINKED ISSUE #6: feat: Read motion sensor

### PR DESCRIPTION
**Pull request by joe-sacco**

### CODE REVIEW

File: src/sensor_motion.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import RPi.GPIO as GPIO
+from time import sleep
+
+GPIO.setmode(GPIO.BCM)
+GPIO.setwarnings(False)
+pir = 16
+GPIO.setup(pir, GPIO.IN)
+
+print('Initializing Sensor...')
+sleep(2)
```
Comment (Takashicc) [inline src/sensor_motion.py]: Is `sleep` necessary?

File: src/sensor_motion.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        # motion detected
+        return True
+    
+    elif not GPIO.input(pir):
+        # no motion detected
+        return False
+
+    elif KeyboardInterrupt: 
+        GPIO.cleanup() # reset GPIO
+        print("Program ended")
```
Comment (Takashicc) [inline src/sensor_motion.py]: Wrap with the `try-catch`.
Store the `GPIO.input(dir)` to the variable.

Change `elif KeyboardInterrupt` block to `except KeyboardInterrupt`.
```python
elif KeyboardInterrupt: 
    GPIO.cleanup() # reset GPIO
    print("Program ended")
```

Also, Is there any chance that `GPIO.input` will throw an error?

**Review by Takashicc — CHANGES_REQUESTED** [review]

File: src/sensor_motion.py (hunk 1 above)
Comment (joe-sacco) [reply src/sensor_motion.py]: Yes

Comment (Takashicc) [reply src/sensor_motion.py]: Why? If you remove the `sleep` is it going to cause an error?

File: src/sensor_motion.py (hunk 2 above)
Comment (joe-sacco) [reply src/sensor_motion.py]: I don't know if GPIO.input could throw an error, but I assume its possible

File: src/sensor_motion.py [hunk 3]
```diff
@@ -7,17 +7,24 @@
...
+        elif not motion:
+            # print("<no motion>")
+            return False
+
+    except KeyboardInterrupt:
+        GPIO.cleanup() #reset GPIO
+
+while True:
+    print(getMotion())
+    sleep(2)
```
Comment (Takashicc) [inline src/sensor_motion.py]: If this is for testing purpose, can you delete this?

**Review by Takashicc — CHANGES_REQUESTED** [review]

File: src/sensor_motion.py (hunk 1 above)
Comment (joe-sacco) [reply src/sensor_motion.py]: It takes time for the sensor to initialize so we need to include the sleep to give the hardware time, then run the code.

**Review by Takashicc — APPROVED** [review]
LGTM!

**Review by Takashicc — APPROVED** [review]

**Pull request merged**